lib/dry/validation/schema.rb

```diff
@@ -92,6 +94,14 @@ def [](name)
       def predicates
         self.class.predicates
       end
+
+      private
+
+      def validate_input(input)
+        unless input.is_a?(Hash)
+          raise ArgumentError, "expected a Hash but received +#{input.inspect}+"
+        end
```

Comment:
We could introduce `is_a?` `instance_of?` and `kind_of?` for this purpose
and you could always have a custom one too
On Fri, 25 Dec 2015 at 18:35, Zach Olauson <notifications@github.com> wrote:

> In lib/dry/validation/schema.rb
> <https://github.com/dryrb/dry-validation/pull/32#discussion_r48442309>:
>
> > @@ -92,6 +94,14 @@ def [](name)
> >        def predicates
> >          self.class.predicates
> >        end
> > +
> > +      private
> > +
> > +      def validate_input(input)
> > +        unless input.is_a?(Hash)
> > +          raise ArgumentError, "expected a Hash but received +#{input.inspect}+"
> > +        end
>
> @solnic <https://github.com/solnic> is there a plan to have a similar
> implementation for attribute based validations? Something like:
>
> class Schema < Dry::Validation::Schema
>
>
>   entity? do
>     attr(:name, &:filled?)
>   endend
>
> I doubt entity? is the right name, but just curious. If that was in place
> we could potentially use that precondition on the predicate to determine if
> the input is of the correct type.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dryrb/dry-validation/pull/32/files#r48442309>.
>